pkg/koordlet/statesinformer/states_noderesourcetopology.go

```diff
@@ -0,0 +1,186 @@
+						BlockOwnerDeletion: &blocker,
+					},
+				},
+			},
+		}
+		_, err = s.topologyClient.TopologyV1alpha1().NodeResourceTopologies().Create(ctx, topology, metav1.CreateOptions{})
+		if err != nil {
+			klog.Errorf("failed to create node resource topology %s, err: %v", topologyName, err)
+			return
+		}
+	}
+}
+
+func (s *statesInformer) reportNodeTopology() {
+	nodeName := s.node.Name
+	topologyName := fmt.Sprintf("koord-%s", nodeName)
+	ctx := context.TODO()
+	nodeCpuInfo, err := s.metricsCache.GetNodeCPUInfo(&metriccache.QueryParam{})
+	if err != nil {
+		klog.Errorf("failed to get node cpu info ")
+		return
+	}
+
+	cpuUsage := make(map[int32]*extension.CPUInfo)
+	cpuTopology := &extension.CPUTopology{}
+	for _, cpu := range nodeCpuInfo.ProcessorInfos {
+		info := extension.CPUInfo{
+			ID:     cpu.CPUID,
+			Core:   cpu.CoreID,
+			Socket: cpu.SocketID,
+			Node:   cpu.NodeID,
+		}
+		cpuTopology.Detail = append(cpuTopology.Detail, info)
+		cpuUsage[cpu.CPUID] = &info
+	}
+	cpuTopologyJson, err := json.Marshal(cpuTopology)
+	if err != nil {
+		klog.Errorf("failed to marshal cpu topology of node %s, err: %v", nodeName, err)
+		return
+	}
+
+	podAllocs := []extension.PodCPUAlloc{}
+	podMetas := s.GetAllPods()
+	for _, podMeta := range podMetas {
+		status, err := extension.GetResourceStatus(podMeta.Pod.Annotations)
+		if err != nil {
+			klog.Errorf("failed to get resource status of pod %s, err: %v", podMeta.Pod.Name, err)
+			continue
+		}
+		if status.CPUSet != "" {
+			podAllocs = append(podAllocs, extension.PodCPUAlloc{
+				Name:      podMeta.Pod.Name,
+				Namespace: podMeta.Pod.Namespace,
+				UID:       podMeta.Pod.UID,
+				CPUSet:    status.CPUSet,
+			})
+
+			set, err := cpuset.Parse(status.CPUSet)
+			if err != nil {
+				klog.Errorf("failed to parse cpuset info of pod %s, err: %v", podMeta.Pod.Name, err)
+				continue
+			}
+			for _, cpuID := range set.ToSliceNoSort() {
+				cpuUsage[int32(cpuID)] = nil
+			}
+		}
+	}
+
+	// make result stable
+	sort.Sort(extension.PodCPUAllocs(podAllocs))
+
+	podAllocsJson, err := json.Marshal(podAllocs)
+	if err != nil {
+		klog.Errorf("failed to marshal pod alloc of node %s, err: %v", nodeName, err)
+		return
+	}
+
+	// nodeID -> cpulist
+	nodeIDToCpus := make(map[int32][]int)
+	for cpuID, info := range cpuUsage {
+		if info != nil {
+			nodeIDToCpus[info.Node] = append(nodeIDToCpus[info.Node], int(cpuID))
+		}
+	}
+
+	sharePools := []extension.CPUSharedPool{}
+	for nodeID, cpus := range nodeIDToCpus {
+		if len(cpus) <= 0 {
+			continue
+		}
+		set := cpuset.NewCPUSet(cpus...)
+		sharePools = append(sharePools, extension.CPUSharedPool{
+			CPUSet: set.String(),
+			Node:   nodeID,
+			Socket: cpuUsage[int32(cpus[0])].Socket,
+		})
+	}
+
+	cpuSharePoolsJson, err := json.Marshal(sharePools)
+	if err != nil {
+		klog.Errorf("failed to marshal cpushare pools of node %s, err: %v", nodeName, err)
+		return
+	}
+
+	err = retry.RetryOnConflict(retry.DefaultBackoff, func() error {
```

Comment:
Conflicts generally do not occur, but other errors may be encountered, such as `TooManyRequests`